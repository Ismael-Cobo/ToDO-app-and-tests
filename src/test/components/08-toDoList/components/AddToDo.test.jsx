const { shallow } = require("enzyme")
const { AddToDo } = require("../../../../components/08-toDoList/components/AddToDo")

describe('prueba para el componente <AddToDo />', () => {

    const handleAddToDo = jest.fn()

    const wrapper = shallow(<AddToDo handleAddToDo={handleAddToDo} />)

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('no debe de llamar al handleAddToDo', () => {

        const formulario = wrapper.find('.todo__form')

        formulario.simulate('submit', {preventDefault(){}})

        expect(handleAddToDo).toHaveBeenCalledTimes(0)

    })

    test('debe de llamar al handleAddToDo', () => {

        const value = 'hola que tal?'

        wrapper.find('.form__input').simulate('change', { target:{value: value }})

        const formulario = wrapper.find('.todo__form')
        
        formulario.simulate('submit', {preventDefault(){}})

        expect(handleAddToDo).toHaveBeenCalledTimes(1)
        expect(handleAddToDo).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false
        })

        //Para comprobar que se hizo el reset
        expect(wrapper.find('.form__input').prop('value')).toBe('')

    })



})
