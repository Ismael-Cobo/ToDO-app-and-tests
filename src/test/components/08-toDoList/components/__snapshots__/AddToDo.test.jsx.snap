// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba para el componente <AddToDo /> debe de mostrarse correctamente 1`] = `
<form
  className="todo__form"
  onSubmit={[Function]}
>
  <label
    className="form__title"
    htmlFor="todo"
  >
     Today I need to 
  </label>
  <div
    className="form__input__group"
  >
    <input
      className="form__input"
      id="todo"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="form__btn"
      type="submit"
      value="submit"
    />
  </div>
</form>
`;
